File: JVM/jnetcall-java-stdio/src/main/java/jnetcall/java/server/ServiceHosts.java
package jnetcall.java.server;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;
import java.io.PrintStream;

import jnetbase.java.meta.Reflect;
import jnetbase.java.threads.ThreadExecutor;
import jnetcall.java.api.io.ISendTransport;
import jnetcall.java.common.ByteMarks;
import jnetcall.java.impl.enc.BinaryEncoding;
import jnetcall.java.impl.io.StreamTransport;

public final class ServiceHosts {

    public static <T> ClassHosting create(Class<T> serviceClass) throws IOException {
        InputStream stdIn = System.in;
        PrintStream stdOut = System.out;
        ByteMarks.writeSync(stdIn, stdOut);
        ISendTransport protocol = initDefault(stdIn, stdOut);
        return create(serviceClass, protocol);
    }

    private static ISendTransport initDefault(InputStream stdIn, OutputStream stdOut) {
        BinaryEncoding enc = new BinaryEncoding();
        return new StreamTransport(enc, stdIn, stdOut);
    }

    private static <T> ClassHosting create(Class<T> serviceClass, ISendTransport protocol) {
        T instance = Reflect.createNew(serviceClass);
        ThreadExecutor pool = new ThreadExecutor();
        ClassHosting host = new ClassHosting(instance, protocol, pool);
        return host;
    }
}

File: NET/JNetCall.Sharp.StdIO/Server/ServiceHosts.cs
using System;
using System.IO;
using JNetBase.Sharp.Threads;
using JNetCall.Sharp.API.IO;
using JNetCall.Sharp.Common;
using JNetCall.Sharp.Impl.Enc;
using JNetCall.Sharp.Impl.IO;

namespace JNetCall.Sharp.Server
{
    public static class ServiceHosts
    {
        public static ClassHosting Create<T>() where T : new()
        {
            var stdIn = Console.OpenStandardInput();
            var stdOut = Console.OpenStandardOutput();
            ByteMarks.WriteSync(stdIn, stdOut);
            var protocol = InitDefault(stdIn, stdOut);
            return Create<T>(protocol);
        }

        private static ISendTransport InitDefault(Stream stdIn, Stream stdOut)
        {
            var enc = new BinaryEncoding();
            return new StreamTransport(enc, stdIn, stdOut);
        }

        private static ClassHosting Create<T>(ISendTransport protocol) where T : new()
        {
            var instance = new T();
            var pool = new ThreadExecutor();
            var host = new ClassHosting(instance, protocol, pool);
            return host;
        }
    }
}